packages/@aws-cdk/aws-rds/lib/engine.ts

```diff
@@ -0,0 +1,53 @@
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import { compare } from './private/version';
+import { ParameterGroupFamily } from './props';
+
+/** @internal */
+export abstract class BaseEngine {
+  /**
+   * The engine.
+   */
+  public readonly name: string;
+
+  /**
+   * The single user secret rotation application.
+   */
+  public readonly singleUserRotationApplication: secretsmanager.SecretRotationApplication;
+
+  /**
+   * The multi user secret rotation application.
+   */
+  public readonly multiUserRotationApplication: secretsmanager.SecretRotationApplication;
+
+  private readonly parameterGroupFamilies?: ParameterGroupFamily[];
+
+  constructor(
+    name: string,
+    singleUserRotationApplication: secretsmanager.SecretRotationApplication,
+    multiUserRotationApplication: secretsmanager.SecretRotationApplication,
+    parameterGroupFamilies?: ParameterGroupFamily[]) {
+
+    this.name = name;
+    this.singleUserRotationApplication = singleUserRotationApplication;
+    this.multiUserRotationApplication = multiUserRotationApplication;
+    this.parameterGroupFamilies = parameterGroupFamilies;
+  }
+
+  protected calculateParameterGroupForEngineVersion(engineVersion: string | undefined) {
```

Comment:
crazy long name. can this be terser please. why not keep the same as `parameterGroupFamily()` or some slight variant if it conflicts with the existing public API in `DatabaseClusterEngine`?

Reply:
Looks like this is only used in `DatabaseClusterEngine`, should we just move it there?

Reply:
OK, done.